apps/auto-email-sender-new/src/index.ts

```diff
@@ -1,15 +1,107 @@
-import dotenv from 'dotenv'
 import { rolesRenderer } from './roles-renderer'
 import { subsToQueue } from './subs-to-queue'
-import { assignRoles } from './roles-assigner'
+import { assignRoles } from './roles-assigner-new'
 import { emailPreRender } from './email-pre-render'
 import { composeEmail } from './email-composer'
 import { emailSender } from './email-sender'
+import { getSupabaseClient } from 'db'
+import { Parser } from '@json2csv/plainjs'
+import fs from 'fs'
+import { exec } from 'child_process'
+import { promisify } from 'util'
+import path from 'path'
+import axios from 'axios'
 
-dotenv.config()
+const generateCsv = async (data: any, filePath: string) => {
+  const parser = new Parser()
+  const csv = parser.parse(data.data)
+  const absolutePath = path.resolve(__dirname, filePath)
+  fs.writeFileSync(absolutePath, csv)
+}
+
+const runCommand = promisify(exec)
+
+const checkMatchRolesUp = async (
+  url: string,
+  maxAttempts = 10,
+  interval = 5000
+) => {
+  const delay = (ms: number) =>
+    new Promise((resolve) => setTimeout(resolve, ms))
```

Comment:
Don't redeclare the `delay` function at each call of `generateCsv`, move it to some utils file or to a single file of it's own.